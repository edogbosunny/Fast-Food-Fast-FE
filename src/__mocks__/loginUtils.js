/* eslint-disable import/prefer-default-export */
export const props = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
  user: {},
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
  location: {
    hash: '',
    pathname: '/login',
    search: '',
  },
  loginUser: jest.fn(),
  match: { path: '/login', url: '/login', isExact: true },
};

export const mockedLoginState = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
};

export const nextProps = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
};

export const Loginstate = {
  email: '',
  password: '',
};
