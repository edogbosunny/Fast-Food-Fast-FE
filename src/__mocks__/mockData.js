const mock = {
  loginDetails: {
    email: 'test@test.com',
    password: '123456',
  },
  registerDetails: {
    email: 'test1@test.com',
    password: 123456,
    confirmpassword: 123456,
    firstname: 'test',
    lastname: 'test,',
  },
  errRegisterDetails: {
    email: 'test1@test.com',
    password: 123456,
    confirmpassword: 123456,
    firstname: 'test',

  },
};
export default mock;
