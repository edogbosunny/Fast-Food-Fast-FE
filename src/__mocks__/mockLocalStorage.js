let localStorage = {};

export default {
  setItem(key, value) {
    return Object.assign(localStorage, { [key]: value });
  },
  removeItem(key) {
    return delete localStorage[key];
  },
  getItem(key) {
    return localStorage[key];
  },
  clear() {
    localStorage = {};
  },
};
