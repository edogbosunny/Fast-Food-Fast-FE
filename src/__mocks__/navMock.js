/* eslint-disable import/prefer-default-export */
export const props = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
  location: {
    hash: '',
    pathname: '/login',
    search: '',
  },
  logoutUser: jest.fn(),
  match: { path: '/login', url: '/login', isExact: true },
};

export const mockState = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
};
