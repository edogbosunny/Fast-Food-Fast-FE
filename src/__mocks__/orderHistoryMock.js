/* eslint-disable import/prefer-default-export */
export const props = {
  dispatch: jest.fn(),
  orderHistory: {
    loading: false,
    orderHistory: {
      count: 30,
      mealItem: [{}],
      message: 'Food order retrieved succesfully',
    },
    getAllOrders: jest.fn(),
    getUserOrderById: jest.fn(),
  },
};

export const tableProps = {
  getAllOrders: jest.fn(),
  getUserOrderById: jest.fn(),
  orderHistory: {
    loading: false,
    orderHistory: {
      count: 30,
      data: [{}],
      message: 'Food order retrieved succesfully',
    },
  },
};
