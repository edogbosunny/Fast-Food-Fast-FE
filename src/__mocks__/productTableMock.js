/* eslint-disable import/prefer-default-export */
export const props = {
  cart: {
    count: 800,
    cart: {
      0: {
        meal_id: 1, price: '600', meal: 'Spaniish Noodles', qty: 1,
      },
    },
    items: null,
    cartCount: 800,
  },
  checkoutOrder: jest.fn(),
  order: {
    order: {},
  },
  history: {
    push: jest.fn(),
  },
  removeFromCart: jest.fn(),
};
