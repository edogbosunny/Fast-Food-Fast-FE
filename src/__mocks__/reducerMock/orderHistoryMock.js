const mock = {
  loading: true,
  orderHistory: {
    message: '',
    count: 0,
    data: [],
  },
};
export default mock;
