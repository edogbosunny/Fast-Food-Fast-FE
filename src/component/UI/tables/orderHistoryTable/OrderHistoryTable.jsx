/* eslint-disable no-unused-vars */
/* eslint-disable import/no-named-as-default */
/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable no-shadow */
/* eslint-disable react/prop-types */
/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import decode from 'jwt-decode';
import '../../../../styles/global.scss';
import '../tables.css';
import OrderHistoryComponent from './OrderHistoryComponent';
import { getAllOrders, getUserOrderById } from '../../../../actions/orderHistoryAction';

export class OrderHistoryTable extends Component {
  componentDidMount() {
    const { getAllOrders, getUserOrderById } = this.props;
    const token = localStorage.getItem('jwtToken');
    const decoded = decode(token);
    localStorage.setItem('userRole', decoded.userRole);
    localStorage.setItem('userId', decoded.id);
    const userRole = localStorage.getItem('jwtToken');
    if (userRole === 'admin') {
      getAllOrders();
    } else {
      getUserOrderById();
      // getAllOrders();
    }
  }

  render() {
    const { orderHistory: { orderHistory } } = this.props;
    // console.log('orderhistory-->', this.props);
    return (
      <OrderHistoryComponent />
    );
  }
}

export const mapStateToProps = state => ({
  orderHistory: state.orderHistory,
  loading: state.loader,
});
export default connect(mapStateToProps, { getAllOrders, getUserOrderById })(OrderHistoryTable);
