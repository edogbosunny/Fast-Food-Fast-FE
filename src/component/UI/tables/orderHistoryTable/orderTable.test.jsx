import React from 'react';
import { shallow } from 'enzyme';
import { tableProps } from '../../../../__mocks__/orderHistoryMock';
import { OrderHistoryTable, mapStateToProps } from './OrderHistoryTable';

describe('orderHistore component test-suit', () => {
  it('should match snapshot', () => {
    localStorage.setItem('jwtToken', 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NCwidXNlclJvbGUiOiJ1c2VyIiwiaWF0IjoxNTQ5NTIyNzYwLCJleHAiOjE1NDk2MDkxNjB9.CI4Tg7J87RH-fOaqjZ4l6JsE4Jjl-Ai1gtkPtiJqMbo');
    localStorage.setItem('userId', 1);
    localStorage.setItem('userRole', 'user');
    const wrapper = shallow(<OrderHistoryTable {...tableProps} />);
    expect(wrapper).toMatchSnapshot();
    wrapper.debug();
  });

  xit('should test mapStateToProps', () => {
    const state = mapStateToProps(mapStateToProps);
    expect(state).toBeTruthy();
  });
});
