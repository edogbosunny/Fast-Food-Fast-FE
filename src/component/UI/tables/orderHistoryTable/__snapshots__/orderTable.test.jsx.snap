// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orderHistore component test-suit should match snapshot 1`] = `<Connect(OrderHistoryComponent) />`;
